pkg/asset/machines/aws/machines.go

```diff
@@ -13,19 +16,19 @@ import (
 	"k8s.io/utils/pointer"
 	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1beta1"
 
+	"github.com/openshift/installer/pkg/asset/installconfig"
 	"github.com/openshift/installer/pkg/types"
 	"github.com/openshift/installer/pkg/types/aws"
 )
 
 // Machines returns a list of machines for a machinepool.
-func Machines(clusterID string, config *types.InstallConfig, pool *types.MachinePool, osImage, role, userDataSecret string) ([]machineapi.Machine, error) {
-	if configPlatform := config.Platform.Name(); configPlatform != aws.Name {
+func Machines(clusterID string, installConfig *installconfig.InstallConfig, pool *types.MachinePool, osImage, role, userDataSecret string) ([]machineapi.Machine, error) {
```

Comment:
there are external users of this function _like hive_, so using the asset is not appropriate here,

Reply:
So i'm fine if we don't keep the best compatibility, but requiring the asset is probably non-starter.

Would rather see the subnet <> zone passed. maybe list of subnets exactly indexed for zones...

Reply:
> Would rather see the subnet <> zone passed. maybe list of subnets exactly indexed for zones...

Done and rebased onto master with 80a252193 -> 06817a106.